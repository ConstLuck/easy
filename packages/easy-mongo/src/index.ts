export * from './MongoGateway';
export * from './MongoProvider';
