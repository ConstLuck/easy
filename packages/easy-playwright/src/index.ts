export * from './PlaywrightElement';
export * from './PlaywrightTester';
