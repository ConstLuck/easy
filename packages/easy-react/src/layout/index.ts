export * from './For';
export * from './If';
