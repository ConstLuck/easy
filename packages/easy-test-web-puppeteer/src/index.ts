export * from './PuppeteerElement';
export * from './PuppeteerTester';
