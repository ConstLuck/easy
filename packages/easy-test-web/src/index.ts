export * from './Tester';
export * from './ElementTester';
