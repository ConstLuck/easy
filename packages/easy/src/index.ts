export * from './data';
export * from './domain';
export * from './http';
export * from './health';
export * from './mongo';
export * from './process';
export * from './resources';
export * from './services';
export * from './security';
export * from './sql';
export * from './types';
export * from './utils';
export * from './validation';
