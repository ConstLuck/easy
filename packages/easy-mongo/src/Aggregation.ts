import { ifDefined, isPresent, Optional } from '@thisisagile/easy';
import { Filter, FindOptions } from './MongoProvider';
import { toMongoType } from './Utils';

export const asc = 1;
export const desc = -1;

export const aggregation = {
  match: (f: Filter) => ({ $match: f }),
  id: (id: Filter) => aggregation.match({ id }),
  eq: (key: string, value: Filter) => aggregation.match({ [key]: value }),
  gt: (key: string, value: Filter) => ({ [key]: { $gt: value } }),
  gte: (key: string, value: Filter) => ({ [key]: { $gte: value } }),
  lt: (key: string, value: Filter) => ({ [key]: { $lt: value } }),
  lte: (key: string, value: Filter) => ({ [key]: { $lte: value } }),
  sum: (to: string, from: string = to) => ({ [to]: { $sum: `$${from}` } }),
  count: (to = 'count') => ({ [to]: { $count: {} } }),
  after: (key: string, date: unknown) => aggregation.match(aggregation.gt(key, toMongoType(date))),
  group: (by: Filter, ...fs: Filter[]) => ({ $group: Object.assign({ _id: by }, ...fs) }),
  skip: ({ skip: $skip }: FindOptions): Optional<Filter> => ifDefined($skip, { $skip }),
  take: ({ take: $limit }: FindOptions): Optional<Filter> => ifDefined($limit, { $limit }),
  sort: ($sort: Record<string, typeof asc | typeof desc>): Optional<Filter> => (isPresent($sort) ? { $sort } : undefined),
  asc: (key: string) => aggregation.sort({ [key]: asc }),
  desc: (key: string) => aggregation.sort({ [key]: desc }),
  date: (key: string, format = '%Y-%m-%d') => ({ $dateToString: { date: `$${key}`, format } }),
};
