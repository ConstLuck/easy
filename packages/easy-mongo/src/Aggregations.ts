import { ifDefined, isPresent, Optional } from '@thisisagile/easy';
import { Filter, FindOptions } from './MongoProvider';
import { toMongoType } from './Utils';

export const asc = 1;
export const desc = -1;

export const aggregations = {
  gt: (key: string, value: Filter) => ({ [key]: { $gt: value } }),
  gte: (key: string, value: Filter) => ({ [key]: { $gte: value } }),
  lt: (key: string, value: Filter) => ({ [key]: { $lt: value } }),
  lte: (key: string, value: Filter) => ({ [key]: { $lte: value } }),
  match: (f: Filter) => ({ $match: f }),
  id: (id: Filter) => aggregations.match({ id }),
  eq: (key: string, value: Filter) => aggregations.match({ [key]: value }),
  after: (key: string, date: unknown) => aggregations.match(aggregations.gte(key, toMongoType(date))),
  before: (key: string, date: unknown) => aggregations.match(aggregations.lt(key, toMongoType(date))),
  sum: (to: string, from: string = to) => ({ [to]: { $sum: `$${from}` } }),
  count: (to = 'count') => ({ [to]: { $count: {} } }),
  groupBy: (by: Filter) => ({
    and: (...fs: Filter[]) => ({ $group: Object.assign({ _id: by }, ...fs) }),
  }),
  skip: ({ skip: $skip }: FindOptions): Optional<Filter> => ifDefined($skip, { $skip }),
  take: ({ take: $limit }: FindOptions): Optional<Filter> => ifDefined($limit, { $limit }),
  sort: ($sort: Record<string, typeof asc | typeof desc>): Optional<Filter> => (isPresent($sort) ? { $sort } : undefined),
  asc: (key: string) => aggregations.sort({ [key]: asc }),
  desc: (key: string) => aggregations.sort({ [key]: desc }),
  date: (key: string, format = '%Y-%m-%d') => ({ $dateToString: { date: `$${key}`, format } }),
};
