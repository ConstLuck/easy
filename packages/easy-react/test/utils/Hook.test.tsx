import React, { useEffect } from 'react';
import '@thisisagile/easy-test';
import { rendersWait } from '@thisisagile/easy-test-react';
import { useEntity, useList, useToggle } from '../../src';
import { Address, toList } from '@thisisagile/easy';

const city = 'Amsterdam';

const ToggleHook = () => {
  const [toggle, setToggle] = useToggle(false);
  const [, setVisible] = useToggle();

  useEffect(() => {
    setToggle();
    setVisible();
  }, []);
  return <div id={'42'}>{`${toggle}`}</div>;
};

const EntityHook = () => {
  const [address, setAddress] = useEntity<Address>();

  useEffect(() => {
    setAddress(new Address({ city }));
  }, []);
  return <>{`${address}`}</>;
};

const ListHook = () => {
  const [list, setList] = useList<Address>();

  useEffect(() => {
    setList(toList<Address>(new Address({ city })));
  }, []);
  return <>{`${list.first()}`}</>;
};

describe('Hooks', () => {
  test('component with useToggle hook renders correctly as default value is false.', async () => {
    const { container, byText } = await rendersWait(<ToggleHook />);
    expect(container).toMatchSnapshot();
    expect(byText('true')).toBeDefined();
  });

  test('component with useEntity hook renders correctly.', async () => {
    const { container, byText } = await rendersWait(<EntityHook />);
    expect(container).toMatchSnapshot();
    expect(byText(city)).toBeDefined();
  });

  test('component with useList hook renders correctly.', async () => {
    const { container, byText } = await rendersWait(<ListHook />);
    expect(container).toMatchSnapshot();
    expect(byText(city)).toBeDefined();
  });
});
