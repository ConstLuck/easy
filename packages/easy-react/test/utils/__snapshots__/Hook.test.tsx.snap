// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hooks component with useEntity hook renders correctly. 1`] = `
<div>
  Amsterdam
</div>
`;

exports[`Hooks component with useList hook renders correctly. 1`] = `
<div>
   Amsterdam
</div>
`;

exports[`Hooks component with useToggle hook renders correctly. 1`] = `
<div>
  <div
    id="42"
  >
    true
  </div>
</div>
`;
